src/request/connect/Connect.js

```diff
@@ -0,0 +1,139 @@
+
+        // Set up password box
+        const $passwordBox = /** @type {HTMLFormElement} */ (document.querySelector('#password-box'));
+        this._passwordBox = new PasswordBox($passwordBox, {
+            hideInput: !request.keyInfo.encrypted,
+            buttonI18nTag: 'passwordbox-connect-account',
+            minLength: request.keyInfo.hasPin ? Key.PIN_LENGTH : undefined,
+        });
+
+        $button.addEventListener('click', () => {
+            $button.classList.add('display-none');
+            $passwordBox.classList.remove('display-none');
+        });
+
+        this._passwordBox.on(
+            PasswordBox.Events.SUBMIT,
+            password => this._onConfirm(request, resolve, reject, password),
+        );
+    }
+
+    /**
+     * @param {Parsed<KeyguardRequest.ConnectRequest>} request
+     * @param {Connect.resolve} resolve
+     * @param {reject} reject
+     * @param {string} [password]
+     * @returns {Promise<void>}
+     * @private
+     */
+    async _onConfirm(request, resolve, reject, password) {
+        TopLevelApi.setLoading(true);
+
+        const passwordBuf = password ? Utf8Tools.stringToUtf8ByteArray(password) : undefined;
+
+        /** @type {Key?} */
+        let key = null;
+        try {
+            key = await KeyStore.instance.get(request.keyInfo.id, passwordBuf);
+        } catch (error) {
+            const errorMessage = error instanceof Error ? error.message : String(error);
+            if (errorMessage === 'Invalid key') {
+                TopLevelApi.setLoading(false);
+                this._passwordBox.onPasswordIncorrect();
+                return;
+            }
+            reject(new Errors.CoreError(errorMessage));
+            return;
+        }
+
+        if (!key) {
+            reject(new Errors.KeyNotFoundError());
+            return;
+        }
+
+        /** @type {KeyguardRequest.SignatureResult[]} */
+        const signatures = [];
+
+        for (const keyPath of request.requestedKeyPaths) {
+            const publicKey = key.derivePublicKey(keyPath);
+
+            /** @type {Uint8Array} */
+            const messageBytes = Utf8Tools.stringToUtf8ByteArray(request.challenge);
+            const signature = key.signMessage(keyPath, messageBytes);
+
+            signatures.push({
+                publicKey: publicKey.serialize(),
+                signature: signature.serialize(),
+            });
+        }
+
+        const rsaPublicCryptoKey = await key.getRsaPublicKey(Key.defaultEncryptionKeyParams);
+        const keyParams = /** @type {RsaKeyPairExport} */ (key.rsaKeyPair).keyParams;
+
+        /** @type {KeyguardRequest.ConnectResult} */
+        const result = {
+            signatures,
+            encryptionKey: {
+                format: 'spki',
+                keyData: new Uint8Array(await window.crypto.subtle.exportKey('spki', rsaPublicCryptoKey)),
+                algorithm: { name: 'RSA-OAEP', hash: 'SHA-256' },
```

Comment:
Can read algorithm from `rsaPublicCryptoKey.algorithm` instead of hard coding the constant here.

```suggestion
                algorithm: {
                    name: rsaPublicCryptoKey.algorithm.name,
                    hash: (/** @type {RsaHashedKeyAlgorithm} */ (rsaPublicCryptoKey.algorithm)).hash.name,
                },
```

Reply:
Resolved in #524